src/eko/evolution_operator/grid.py

```diff
@@ -179,7 +179,11 @@ def get_threshold_operators(self, path):
 
             # Compute the matching conditions and store it
             if len(path) > 1:
```

Comment:
I guess, we should add a `and seg.q2_to not in self._matching_operators` as we did above for caching